pkg/resources/server.go

```diff
@@ -104,21 +108,14 @@ func (rsc *resourcesServerPort) Dial(unixSocketPath string, timeout time.Duratio
 func newResourceServer(config *types.UserConfig, devices []types.PciNetDevice, watcherMode bool, resourcePrefix,
 	socketSuffix string) (types.ResourceServer, error) {
 	var devs []*pluginapi.Device
-	deviceSpec := make([]*pluginapi.DeviceSpec, 0)
 
 	sockDir := activeSockDir
 
 	if config.RdmaHcaMax < 0 {
 		return nil, fmt.Errorf("error: Invalid value for rdmaHcaMax < 0: %d", config.RdmaHcaMax)
 	}
 
-	for _, device := range devices {
-		rdmaDeviceSpec := device.GetRdmaSpec()
-		if len(rdmaDeviceSpec) == 0 {
-			log.Printf("Warning: non-Rdma Device %s\n", device.GetPciAddr())
-		}
-		deviceSpec = append(deviceSpec, rdmaDeviceSpec...)
-	}
+	deviceSpec := utils.GetDevicesSpec(devices)
```

Comment:
why did you move this logic to utils  and not just keep it in this file ?

Reply:
It is used in more than one location so moved to utils, do u prefer to do same as above

Reply:
its still a method that is local to this file, so for now lets keep it here imo